client/client_test.go

```diff
@@ -1186,6 +1156,37 @@ func testPublishNoData(t *testing.T, rootType string, serverManagesSnapshot bool
 	}
 }
 
+// Publishing an uninitialized repo will fail, but initializing and republishing
+// after should succeed
+func TestPublishUninitializedRepo(t *testing.T) {
+	gun := "docker.com/notary"
+	ts := fullTestServer(t)
+	defer ts.Close()
+
+	// uninitialized repo should fail to publish
+	tempBaseDir, err := ioutil.TempDir("", "notary-tests")
+	assert.NoError(t, err)
+	defer os.RemoveAll(tempBaseDir)
+
+	repo, err := NewNotaryRepository(tempBaseDir, gun, ts.URL,
+		http.DefaultTransport, passphraseRetriever)
+	assert.NoError(t, err, "error creating repository: %s", err)
+	err = repo.Publish()
+	assert.Error(t, err)
```

Comment:
I made a [similar test](https://github.com/docker/notary/commit/c40ec37f50d3a3198e1bf8a5a78c8ab995530b20#diff-fce84029d346504e44323c30d21a3808R1434) that checks for the error in my PR as well -- your test is more thorough so feel free to remove mine and just add the error check if you prefer!